OpenEdXMobile/src/main/java/org/edx/mobile/model/iap/IAPFlowData.kt

```diff
@@ -0,0 +1,29 @@
+package org.edx.mobile.model.iap
+
+data class IAPFlowData(
+    var upgradeMode: UpgradeMode = UpgradeMode.NORMAL,
+    var productId: String = "",
+    var basketId: Long = 0,
+    var purchaseToken: String = "",
+    var isVerificationPending: Boolean = false
+) {
+    fun clear() {
+        upgradeMode = UpgradeMode.NORMAL
+        productId = ""
+        basketId = 0
+        purchaseToken = ""
+        isVerificationPending = false
+    }
+
+    enum class UpgradeMode {
+        NORMAL,
+        SILENT;
+
+        fun isSilentMode() = this == SILENT
+    }
+
+    enum class IAPAction {
+        SHOW_FULL_SCREEN_LOADER,
+        PURCHASE_FLOW_COMPLETE
+    }
```

Comment:
Please test the flow that which class is starting/showing the `FullscreenLoaderDialogFragment` because every Subscriber in the back stack is getting the `SHOW_FULL_SCREEN_LOADER` event and the same goes for the `PURCHASE_FLOW_COMPLETE` event.